### PR TITLE
Add Wasmbindings

### PR DESCRIPTION
**Pull request by amityadav0**
- Custom query for GMM
- Custom messages for GMM

### CODE REVIEW

File: wasmbinding/message_plugin.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+func (m *CustomMessenger) swap(ctx sdk.Context, contractAddr sdk.AccAddress, createDenom *bindings.Swap) ([]sdk.Event, [][]byte, error) {
+	err := PerformSwap(m.gmm, m.bank, ctx, contractAddr, createDenom)
+	if err != nil {
+		return nil, nil, errorsmod.Wrap(err, "perform swap token")
+	}
+	return nil, nil, nil
+}
+
+// PerformSwap is used with swap to swap a token denom; validates the msgSwap.
+func PerformSwap(f *gmmkeeper.Keeper, b *bankkeeper.Keeper, ctx sdk.Context, contractAddr sdk.AccAddress, swap *bindings.Swap) error {
```
Comment (DedicatedDev) [inline wasmbinding/message_plugin.go:63]: is it possible to add test for this function?

Comment (amityadav0) [reply wasmbinding/message_plugin.go:63]: yes, going to add test for all of the additions

Comment (DedicatedDev) [reply wasmbinding/message_plugin.go:63]: also, please check the lint issue.

Comment (amityadav0) [reply wasmbinding/message_plugin.go:63]: ok

File: wasmbinding/message_plugin.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+var _ wasmkeeper.Messenger = (*CustomMessenger)(nil)
+
+// DispatchMsg executes on the contractMsg.
+func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, contractIBCPortID string, msg wasmvmtypes.CosmosMsg) ([]sdk.Event, [][]byte, error) {
+	if msg.Custom != nil {
+		// only handle the happy path where this is really creating / minting / swapping ...
+		// leave everything else for the wrapped version
+		var contractMsg bindings.OsmosisMsg
+		if err := json.Unmarshal(msg.Custom, &contractMsg); err != nil {
+			return nil, nil, errorsmod.Wrap(err, "osmosis msg")
```
Comment (liangping) [inline wasmbinding/message_plugin.go:44]: please update the error message

Comment (amityadav0) [reply wasmbinding/message_plugin.go]: done

File: wasmbinding/message_plugin.go (hunk 1 above)
Comment (amityadav0) [reply wasmbinding/message_plugin.go]: fixed lint, have added a task for adding tests

**Pull request merged**